bot/mc.go

```diff
@@ -4,43 +4,116 @@ import (
 	"fmt"
 	"log"
 	"os"
+	"time"
 	"venova/sshcmd"
 
 	"github.com/gorcon/rcon"
 )
 
-func restartMinecraft() {
+func manageMinecraftCmd(ctx CommandCtx) error {
+	msg := ctx.Message.Message
+	sess := ctx.Session
+	args := ctx.Args
+	actions := map[string]struct {
+		Message string
+		Command string
+	}{
+		"up":      {"The Minecraft Server has been brought up", "up -d"},
+		"down":    {"The Minecraft server has been brought down", "down"},
+		"restart": {"The Minecraft server has been restarted.", "restart"},
+	}
+	if action, exists := actions[args[0]]; exists {
+		if memberHasRole(msg.Member, mcRoleId) || msg.Author.ID == morthisId {
+			mcMsg, _ := sess.ChannelMessageSend(msg.ChannelID, "Attempting to modify the minecraft server...")
+
+			go func() {
+				err := execDockerCompose(action.Command)
+				if err != nil {
+					log.Printf("%s", err)
+					ctx.Reply(fmt.Sprintf("%s", err))
+					return
+				}
+				time.Sleep(time.Second * 5)
+				sess.ChannelMessageEdit(msg.ChannelID, mcMsg.ID, action.Message)
+			}()
+			return nil
+		}
+	} else {
+		ctx.Reply("Action is not available")
+		return nil
+	}
+	ctx.Reply("You're not a minecraft administrator.")
+	return nil
+}
+
+func mcCmd(ctx CommandCtx) error {
+	m := ctx.Message
+	if m.Author.ID == blueId || m.Author.ID == morthisId {
+		res, err := minecraftCommand(ctx.Args[0])
+		if err != nil {
+			log.Printf("Err: %s", err)
+			return ctx.Reply("Could not send command, Minecraft might be offline.")
+		}
+		return ctx.Reply(res)
+	}
+	return ctx.Reply("Sorry you're not a Minecraft server admin.")
+
+}
+
+func execDockerCompose(action string) error {
 	client, err := sshcmd.ConnectToDev()
 	if err != nil {
-		log.Printf("SSH error: %v", err)
-		return
+		return fmt.Errorf("ssh error: %w", err)
 	}
 	defer client.Close()
 
-	com, err := sshcmd.RunCommand(client, "docker-compose -f /app/docker-compose.yml restart")
+	com, err := sshcmd.RunCommand(client, fmt.Sprintf("docker-compose -f /app/docker-compose.yml %s", action))
 	if err != nil {
-		log.Println("Error:", err.Error())
+		return fmt.Errorf("error running ssh command: %w", err)
 	}
 	log.Printf("Compose command response: %s", com)
+	return nil
 }
 
 func minecraftCommand(command string) (string, error) {
-
 	rconHost := os.Getenv("MC_HOST")
 	rconPort := os.Getenv("RCON_PORT")
 	rconPass := os.Getenv("RCON_PASS")
 
+	if rconHost == "" || rconPort == "" || rconPass == "" {
```

Comment:
Might be nice to do this at startup in the future, either via [an init function](https://go.dev/doc/effective_go#init) or just within our `main.go` initialization we could call `setupMinecraft()` or the likes.